src/Psalm/Internal/Codebase/Analyzer.php

```diff
@@ -229,7 +229,7 @@ private function getFileAnalyzer(
         string $file_path,
         array $filetype_analyzers
     ): FileAnalyzer {
-        $extension = (string) (pathinfo($file_path)['extension'] ?? '');
+        $extension = pathinfo($file_path, PATHINFO_EXTENSION);
```

Comment:
This is more clear. Also it doesn't need the array access, the null coalesce nor the cast